packages/auth/amplify_auth_cognito_dart/lib/src/model/auth_configuration.dart

```diff
@@ -18,6 +18,56 @@ class AuthConfiguration with AWSEquatable<AuthConfiguration> {
     this.pinpointConfig,
   });
 
+  /// Parses [amplifyOutputs] into the appropriate [AuthConfiguration] type.
+  ///
+  /// {@macro amplify_auth_cognito.auth_configuration}
+  factory AuthConfiguration.fromAmplifyOutputs(AmplifyOutputs amplifyOutputs) {
+    final authOutputs = amplifyOutputs.auth;
+    if (authOutputs == null) {
+      throw ConfigurationError(
+        'Invalid config: No Auth config found',
+      );
+    }
+    CognitoUserPoolConfig? userPoolConfig;
```

Comment:
What is the long term plan for the types that are part of AmplifyConfig? Ex: CognitoUserPoolConfig is part of CognitoPluginConfig which is part of AuthConfig.